gcloud-java-core/src/main/java/com/google/gcloud/BasePage.java

```diff
@@ -67,10 +67,10 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof BaseListResult)) {
+    if (!(obj instanceof BasePage)) {
```

Comment:
Given that we only have one Page implementation now, does the name BasePage still make sense?  Or should it be something else like PageImpl?